generators/csharp/codegen/src/ast/Class.ts

```diff
@@ -63,26 +68,30 @@ export declare namespace Class {
 
     interface Operator {
         /* The type of the method */
-        type: "implicit" | "explicit";
+        type: "implicit" | "explicit" | "==" | "!=";
         /* The parameters of the method */
         parameters: Parameter[];
         /* The body of the operator */
         body: CodeBlock;
+        return?: Type;
+        useExpressionBody?: boolean;
     }
 }
 
 export class Class extends AstNode {
     public readonly name: string;
     public readonly namespace: string;
     public readonly access: Access;
+    public readonly static_: boolean;
     public readonly abstract: boolean;
     public readonly sealed: boolean;
+    public readonly readonly: boolean;
     public readonly partial: boolean;
     public readonly reference: ClassReference;
     public readonly parentClassReference: AstNode | undefined;
     public readonly interfaceReferences: ClassReference[];
     public readonly isNestedClass: boolean;
-    public readonly record: boolean;
+    public readonly type: "class" | "record" | "struct" | "record struct";
```

Comment:
Given that we define this enum in two places (in the same file), let's create a type for it and reference it in both places.